feature_asmap.py intermittent fail (                                   AssertionError: [node 0] Expected messages "['CheckAddrman: new 2, tried 2, total 4 started', 'CheckAddrman: completed']" does not partially match log:)

https://cirrus-ci.com/task/5452911883321344?logs=ci#L4941

```
 test  2024-03-12T09:55:01.911000Z TestFramework (ERROR): Assertion failed 
                                   Traceback (most recent call last):
                                     File "/ci_container_base/ci/scratch/build/bitcoin-x86_64-pc-linux-gnu/test/functional/test_framework/test_framework.py", line 131, in main
                                       self.run_test()
                                     File "/ci_container_base/ci/scratch/build/bitcoin-x86_64-pc-linux-gnu/test/functional/feature_asmap.py", line 132, in run_test
                                       self.test_asmap_interaction_with_addrman_containing_entries()
                                     File "/ci_container_base/ci/scratch/build/bitcoin-x86_64-pc-linux-gnu/test/functional/feature_asmap.py", line 90, in test_asmap_interaction_with_addrman_containing_entries
                                       with self.node.assert_debug_log(
                                     File "/usr/lib/python3.10/contextlib.py", line 142, in __exit__
                                       next(self.gen)
                                     File "/ci_container_base/ci/scratch/build/bitcoin-x86_64-pc-linux-gnu/test/functional/test_framework/test_node.py", line 490, in assert_debug_log
                                       self._raise_assertion_error('Expected messages "{}" does not partially match log:\n\n{}\n\n'.format(str(expected_msgs), print_log))
                                     File "/ci_container_base/ci/scratch/build/bitcoin-x86_64-pc-linux-gnu/test/functional/test_framework/test_node.py", line 188, in _raise_assertion_error
                                       raise AssertionError(self._node_msg(msg))
                                   AssertionError: [node 0] Expected messages "['CheckAddrman: new 2, tried 2, total 4 started', 'CheckAddrman: completed']" does not partially match log:
                                    - 2024-03-12T09:53:41.898810Z [http] [httpserver.cpp:306] [http_request_cb] [http] Received a POST request for / from 127.0.0.1:33584
                                    - 2024-03-12T09:53:41.898867Z [httpworker.2] [rpc/request.cpp:187] [parse] [rpc] ThreadRPCServer method=getnodeaddresses user=__cookie__
                                    - 2024-03-12T09:53:41.898904Z [httpworker.2] [logging/timer.h:58] [Log] [addrman] CheckAddrman: new 3, tried 1, total 4 started
                                    - 2024-03-12T09:53:41.899039Z [httpworker.2] [logging/timer.h:58] [Log] [addrman] CheckAddrman: completed (0.12ms)
                                    - 2024-03-12T09:53:41.899054Z [httpworker.2] [addrman.cpp:842] [GetAddr_] [addrman] GetAddr returned 1 random addresses
                                    - 2024-03-12T09:53:41.899065Z [httpworker.2] [logging/timer.h:58] [Log] [addrman] CheckAddrman: new 3, tried 1, total 4 started
                                    - 2024-03-12T09:53:41.899182Z [httpworker.2] [logging/timer.h:58] [Log] [addrman] CheckAddrman: completed (0.11ms)
                                    - 2024-03-12T09:54:26.770784Z [scheduler] [net.cpp:2337] [StartExtraBlockRelayPeers] [net] enabling extra block-relay-only peers

--- a/test/functional/feature_asmap.py
+++ b/test/functional/feature_asmap.py
@@ -39,7 +39,8 @@ def expected_messages(filename):
 class AsmapTest(BitcoinTestFramework):
     def set_test_params(self):
         self.num_nodes = 1
-        self.extra_args = [["-checkaddrman=1"]]  # Do addrman checks on all operations.
+        # Do addrman checks on all operations and use deterministic addrman
+        self.extra_args = [["-checkaddrman=1", "-test=addrman"]]
 
     def fill_addrman(self, node_id):
         """Add 2 tried addresses to the addrman, followed by 2 new addresses."""

--- a/test/functional/rpc_net.py
+++ b/test/functional/rpc_net.py
@@ -319,7 +319,9 @@ class NetTest(BitcoinTestFramework):
 
     def test_addpeeraddress(self):
         self.log.info("Test addpeeraddress")
-        self.restart_node(1, ["-checkaddrman=1", "-test=addrman"])
+        # The node has an existing, non-deterministic addrman from a previous test.
+        # Clear it to have a deterministic addrman.
+        self.restart_node(1, ["-checkaddrman=1", "-test=addrman"], clear_addrman=True)
         node = self.nodes[1]
 
         self.log.debug("Test that addpeerinfo is a hidden RPC")

